### PR TITLE
Add GET method to /group/:uuid/principals/

### PR DESCRIPTION
**Pull request by wcmitchell**
Signed-off-by: Chris Mitchell <cmitchel@redhat.com>

### CODE REVIEW

**Comment by wcmitchell** [comment]
Oh, right! I need to do that too. Thanks for the reminder!

File: rbac/management/group/view.py [hunk 1]
```diff
@@ -360,7 +385,8 @@ def principals(self, request, uuid=None):
             if isinstance(resp, dict) and 'errors' in resp:
                 return Response(status=resp['status_code'], data=resp['errors'])
             output = GroupSerializer(resp)
-            return Response(status=status.HTTP_200_OK, data=output.data)
+        elif request.method == 'GET':
+            output = PrincpalSerializer(group.principals.all(), many=True)
```
Comment (coderbydesign) [inline rbac/management/group/view.py]: I'm guessing we'll probably want to have this data be paginated like all other (apart from `/access/`) endpoints?

File: rbac/management/group/view.py [hunk 2]
```diff
@@ -32,6 +32,7 @@
 from management.permissions import GroupAccessPermission
 from management.principal.model import Principal
 from management.principal.proxy import PrincipalProxy
+from management.principal.serializer import PrincpalSerializer
```
Comment (coderbydesign) [inline rbac/management/group/view.py]: Minor type here and below where it's called. 
```suggestion
from management.principal.serializer import PrincipalSerializer
```

File: rbac/management/group/view.py (hunk 1 above)
Comment (wcmitchell) [reply rbac/management/group/view.py]: Yeah, I went back and forth on that. Wasn't sure if it was worth paginating a bare list of Principals, but I can do that, no problem!

**Comment by wcmitchell** [comment]
This worked in local testing, but I'm a bit concerned about the simplicity of the proxy call and how it will behave in real-world situations.

File: rbac/management/group/view.py [hunk 3]
```diff
@@ -360,7 +383,16 @@ def principals(self, request, uuid=None):
             if isinstance(resp, dict) and 'errors' in resp:
                 return Response(status=resp['status_code'], data=resp['errors'])
             output = GroupSerializer(resp)
-            return Response(status=status.HTTP_200_OK, data=output.data)
+            response = Response(status=status.HTTP_200_OK, data=output.data)
+        elif request.method == 'GET':
+            page = self.paginate_queryset(group.principals.all())
+            serializer = PrincipalSerializer(page, many=True)
+            usernames = serializer.data
```
Comment (coderbydesign) [inline rbac/management/group/view.py]: I think this is going to end up being an ordered dictionary here, which is what the `users` value in the [POST payload](https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/principal/proxy.py#L188) will be set to, and it looks like we actually [need a list](https://backoffice-proxy-insights-services.ext.us-east.aws.preprod.paas.redhat.com/docs/#/default/postUsers)?

Comment (wcmitchell) [reply rbac/management/group/view.py]: It has been returning an OrderedDict, but I wasn't sure if it needed that level of object specificity, but I can do a quick translation of the data into a list here.

File: docs/source/specs/openapi.json [hunk 4]
```diff
@@ -525,6 +525,74 @@
...
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of principals attached to group",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
```
Comment (coderbydesign) [inline docs/source/specs/openapi.json]: I think here for the 200 we may just need:
```
          "200": {
            "description": "List of principals attached to group",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/PrincipalPagination"
                }
              }
            }
          }
```
The output in https://editor.swagger.io/ for this as is doesn't match the actual response, as it nests everything in a principal lists.

**Pull request merged**